deep_qa/common/params.py

```diff
@@ -12,6 +31,28 @@ def _get_choice_error_message(value: Any, choices: List[Any], name: str=None) ->
         return '%s not in acceptable choices: %s' % (value, str(choices))
 
 
+def pop_with_logging(params: Dict[str, Any], key: str):
+    """
+    Performs the functionality associated with dict.pop(key) but with parameter
+    logging. This is required because pop_with_default may receive a default value
+    of None, which means we can't check for it not being passed.
+    """
+    value = params.pop(key)
+    logger.param(key + " : " + str(value))
+    return value
+
+
+def pop_with_default(params: Dict[str, Any], key: str, default: Any):
```

Comment:
What if we just merge this method and the above, making `default: Any=None`?  Then you just call it `get_param`, or `pop_param`, or something, and all parameter reads will just use the same function.